src/pages/VisiumQC.tsx

```diff
@@ -1,43 +1,54 @@
 import React, { ChangeEvent, useCallback, useState } from "react";
-import { GetVisiumQcInfoQuery, LabwareFieldsFragment } from "../types/sdk";
+import { GetVisiumQcInfoQuery, SlotMeasurementRequest } from "../types/sdk";
 import AppShell from "../components/AppShell";
 import WorkNumberSelect from "../components/WorkNumberSelect";
 import LabwareScanner from "../components/labwareScanner/LabwareScanner";
 import Heading from "../components/Heading";
 import { objectKeys } from "../lib/helpers";
-import { Select } from "../components/forms/Select";
+import FormikSelect from "../components/forms/Select";
 import OperationCompleteModal from "../components/modal/OperationCompleteModal";
 import Warning from "../components/notifications/Warning";
 import { ClientError } from "graphql-request";
 import { reload } from "../lib/sdk";
-import VisiumQCType from "../components/visiumQC/VisiumQCType";
+import * as Yup from "yup";
+import { Form, Formik } from "formik";
+import { VisiumQCType } from "../components/visiumQC/VisiumQCType";
 
 export enum QCType {
   SLIDE_PROCESSING = "Slide Processing",
+  CDNA_AMPLIFICATION = "cDNA amplification",
+  CDNA_ANALYSIS = "cDNA analysis",
 }
 
 type VisiumQCProps = {
   info: GetVisiumQcInfoQuery;
 };
 
+const validationSchema = Yup.object().shape({
+  workNumber: Yup.string().optional().label("SGP number"),
+  qcType: Yup.string().required().oneOf(Object.values(QCType)).label("QC Type"),
+  barcode: Yup.string().required().label("Barcode"),
+  slotMeasurements: Yup.array()
+    .of(
+      Yup.object().shape({
+        address: Yup.string().required(),
+        name: Yup.string().oneOf(["Cq value", "Concentration"]),
+        value: Yup.string().required(),
+      })
+    )
+    .optional(),
```

Comment:
Have you checked out yup's `when` method? You may be able to make this optional or required depending on which qcType is selected: https://github.com/jquense/yup#mixedwhenkeys-string--arraystring-builder-object--value-schema-schema-schema

Reply:
I will check it out.